### ISSUE TITLE
Deploying Rancher on k3s with traefik has nginx annotations in ingress object

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#26629]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Deploy Rancher on a k3s cluster with default options

**Result:**
The ingress object created by the Rancher Helm Chart creates nginx annotations:
```yaml
apiVersion: v1
items:
- apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      field.cattle.io/publicEndpoints: '[{"addresses":["10.101.100.135"],"port":80,"protocol":"HTTP","serviceName":"cattle-system:rancher","ingressName":"cattle-system:rancher","hostname":"myhost","allNodes":false}]'
      nginx.ingress.kubernetes.io/proxy-connect-timeout: "30"
      nginx.ingress.kubernetes.io/proxy-read-timeout: "1800"
      nginx.ingress.kubernetes.io/proxy-send-timeout: "1800"
      nginx.ingress.kubernetes.io/ssl-redirect: "false"
    creationTimestamp: "2020-04-14T04:43:27Z"
    generation: 1
    labels:
      app: rancher
      chart: rancher-2.4.2
      heritage: Helm
      release: rancher
    name: rancher
    namespace: cattle-system
    resourceVersion: "2477"
    selfLink: /apis/extensions/v1beta1/namespaces/cattle-system/ingresses/rancher
    uid: 49dcaf1e-09ea-425a-912c-78314a9e19b8
  spec:
    rules:
    - host: myhost
      http:
        paths:
        - backend:
            serviceName: rancher
            servicePort: 80
  status:
    loadBalancer:
      ingress:
      - ip: 10.101.100.135
```
When deploying the Helm Chart on k3s & traefik, these annotations should not be added since nginx is not used as the ingress controller. If there are traefik equivalent settings for proxy and redirects, those should be used instead.

**Other details that may be helpful:**
This doesn't seem to cause a noticeable problem, but should be cleaned up.

See https://github.com/rancher/rancher/blob/master/chart/values.yaml#L50-L52

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.4.2
- Installation option (single install/HA):
HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
k3s
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
AWS - m5.large - 8GB RAM, 2vCPU
- Kubernetes version (use `kubectl version`):

```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4+k3s1", GitCommit:"3eee8ac3a1cf0a216c8a660571329d4bda3bdf77", GitTreeState:"clean", BuildDate:"2020-03-25T16:13:25Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4+k3s1", GitCommit:"3eee8ac3a1cf0a216c8a660571329d4bda3bdf77", GitTreeState:"clean", BuildDate:"2020-03-25T16:13:25Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):
NA

**Comment by tedtramonte** [comment]
I ran across this (and a couple others) while trying to troubleshoot the **very noticeable** problem of not being able to connect to the Rancher UI after deploying in a k3s cluster. I think this is fixed by #26870 and made it into v2.4.3-rc4.

**Comment by timowevel1** [comment]
These annotations are still there :D
